backend/crates/eiffelvis_gen/src/event_set.rs

```diff
@@ -0,0 +1,130 @@
+use std::collections::HashMap;
+
+use crate::event_type::{Event, Link, LinkTargets};
+
+#[derive(Default)]
+pub struct EventSetBuilder {
+    links: HashMap<String, Link>,
+    events: HashMap<String, Event>,
+}
+
+#[derive(Default, Debug)]
+pub struct EventSet {
+    pub links: HashMap<String, Link>,
+    pub events: HashMap<String, Event>,
```

Comment:
These should be hidden as its an implementation detail, will have to design a public api